### ISSUE TITLE
Enable NuGet Audit in source-build-reference-packages

### ISSUE THREAD

**Issue by ViktorHofer** [dotnet/source-build#4673]
See https://github.com/dotnet/arcade/issues/15019 for more details

Also see https://github.com/dotnet/source-build-reference-packages/pull/1016 which attempted turning it on.

**Comment by MichaelSimons** [comment]
I don't think this makes sense in SBRP for the actual reference packages.  We have recently changed our stance on vulnerabilities within the reference packages.  See https://github.com/dotnet/source-build/issues/4838.  We are no longer lifting vulnerable versions as there is no vulnerabilities in reference packages.  By lifting packages we have created a maintenance burden with no benefit.  We are also adding another difference in how SB builds the product vs the Microsoft build.  There would be value in enabling nuget audit for the repo's infrastructure like the generate tooling.
